py3status/modules/check_tcp.py

```diff
@@ -29,23 +31,23 @@ class Py3status:
     format = '{host}:{port} {state}'
     host = 'localhost'
     port = 22
+    string_down = 'DOWN'
+    string_up = 'UP'
 
-    def check_tcp(self):
-        response = {'cached_until': self.py3.time_in(self.cache_timeout)}
+    def post_config_hook(self):
+        self.color_on = self.py3.COLOR_UP or self.py3.COLOR_GOOD
+        self.color_off = self.py3.COLOR_DOWN or self.py3.COLOR_BAD
 
+    def check_tcp(self):
         sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         result = sock.connect_ex((self.host, self.port))
+        state = self.string_down if result else self.string_up
```

Comment:
I did it to get less lines. Optimizing. More clean too (imo).

It may be just perfect/enough for this simple/quick `check_tcp` purpose, but for other complicated things, we would use your way. Can fix that if wanted.